boards/layout_hifive1.ld

```diff
@@ -9,7 +9,7 @@ MEMORY {
    * Note that the SRAM address may need to be changed depending on
    * the kernel binary, check for the actual address of APP_MEMORY!
    */
-  FLASH (rx) : ORIGIN = 0x20040040, LENGTH = 32M
+  FLASH (rx) : ORIGIN = 0x20040040, LENGTH = 4M
```

Comment:
AFAIK hifive1-revb and redv both have 4MB of flash only, revA had 16MB I'm not sure where 32 is from?